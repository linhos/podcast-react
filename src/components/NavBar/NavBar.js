import React from 'react';
import {
    Link
} from "react-router-dom";

import styled from '@emotion/styled';

const Nav = styled.nav`
    display: flex;
    flex-direction: row;
`;

const links = styled.a`
    margin-right: 15px;
`;

function NavBar () {

    return(
        <nav className="Nav"> 
            <Link to="/" className="links">Inicio</Link>
            <Link to={`/search/ann`} className="links">Buscar</Link>
            <Link to="/favourites" className="links">Favoritos</Link>
        </nav>
    )
}

export {NavBar};
