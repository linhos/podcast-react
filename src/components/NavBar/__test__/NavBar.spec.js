import React from 'react';
import { render } from '@testing-library/react';
import { NavBar} from '../NavBar';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history'


test('renders learn react link', () => {
    const history = createMemoryHistory();
    const { getByText } = render(
            <Router history={history}>
                <NavBar />
            </Router>
    );
    const linkElement = getByText(/Inicio/i);
    expect(linkElement).toBeInTheDocument();
});


it('renders', () => {
    const history = createMemoryHistory();
    const {asFragment} = render(
        <Router history={history}>
                <NavBar />
            </Router>
    );

    expect(asFragment()).toMatchSnapshot();
})