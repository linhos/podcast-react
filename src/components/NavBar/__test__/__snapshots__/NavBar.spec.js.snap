// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <nav
    class="Nav"
  >
    <a
      class="links"
      href="/"
    >
      Inicio
    </a>
    <a
      class="links"
      href="/search/ann"
    >
      Buscar
    </a>
    <a
      class="links"
      href="/favourites"
    >
      Favoritos
    </a>
    <a
      class="links"
      href="/login"
    >
      Login
    </a>
  </nav>
</DocumentFragment>
`;
